q2cli/tests/test_usage.py

```diff
@@ -0,0 +1,257 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from click.testing import CliRunner
+import pytest
+import unittest
+
+from q2cli.core.usage import CLIUsage, CLIRenderer
+from q2cli.commands import RootCommand
+
+from qiime2.core.testing.util import get_dummy_plugin
+
+
+@pytest.fixture
+def dummy_plugin(monkeypatch):
+    monkeypatch.setenv('QIIMETEST', '')
+    return get_dummy_plugin()
+
+
+params = [
```

Comment:
Regarding test cases - these all look good - can you add a test for _every other usage example_ in the dummy plugin?

https://github.com/qiime2/qiime2/blob/master/qiime2/core/testing/examples.py#L54-L223

These might be good to start with, that way you can TDD your way into any minor tweaks that might need to happen in the templater (for example, the `set` business I mentioned above).

Reply:
The only missing usage example I see is `comments_only`.

Reply:
Perfect, thanks for checking - can you add the `comments_only` case? Thanks!

Reply:
Resolved in d5fc3e7